alibuild_helpers/sync.py

```diff
@@ -67,8 +67,9 @@ def getRetry(self, url, dest=None, returnResult=False, log=True, session=None, p
           reportTime = time.time()
           result = []
 
+          destFp = None
           try:
-            destFp = open(dest+".tmp", "wb") if dest else None
+            if dest: destFp = open(dest+".tmp", "wb")
```

Comment:
Not sure this is an improvement.

Reply:
maybe we should split the opening of the file from the rest of the try catch...

Reply:
I would live this particular change out and continue with the rest of the PR, since to that I have no particular objections.